cli/container.go

```diff
@@ -9,37 +9,53 @@ import (
 	"os"
 	"time"
 
-	"github.com/docker/go-connections/nat"
-
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/api/types/network"
 	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
 )
 
-func createServer(verbose bool, image string, port string, args []string, env []string, name string, volumes []string) (string, error) {
-	log.Printf("Creating server using %s...\n", image)
+func startContainer(verbose bool, config *container.Config, hostConfig *container.HostConfig, networkingConfig *network.NetworkingConfig, containerName string) (string, error) {
 	ctx := context.Background()
+
 	docker, err := client.NewEnvClient()
 	if err != nil {
 		return "", fmt.Errorf("ERROR: couldn't create docker client\n%+v", err)
 	}
-	reader, err := docker.ImagePull(ctx, image, types.ImagePullOptions{})
-	if err != nil {
-		return "", fmt.Errorf("ERROR: couldn't pull image %s\n%+v", image, err)
-	}
-	if verbose {
-		_, err := io.Copy(os.Stdout, reader)
+
+	resp, err := docker.ContainerCreate(ctx, config, hostConfig, networkingConfig, containerName)
+	if client.IsErrImageNotFound(err) {
+		log.Printf("Pulling image %s...\n", config.Image)
+		reader, err := docker.ImagePull(ctx, config.Image, types.ImagePullOptions{})
 		if err != nil {
-			log.Printf("WARNING: couldn't get docker output\n%+v", err)
+			return "", fmt.Errorf("ERROR: couldn't pull image %s\n%+v", config.Image, err)
 		}
-	} else {
-		_, err := io.Copy(ioutil.Discard, reader)
-		if err != nil {
-			log.Printf("WARNING: couldn't get docker output\n%+v", err)
+		if verbose {
+			_, err := io.Copy(os.Stdout, reader)
+			if err != nil {
+				log.Printf("WARNING: couldn't get docker output\n%+v", err)
+			}
+		} else {
+			_, err := io.Copy(ioutil.Discard, reader)
+			if err != nil {
+				log.Printf("WARNING: couldn't get docker output\n%+v", err)
+			}
```

Comment:
We could add 
```Go
resp, err = docker.ContainerCreate(ctx, config, hostConfig, networkingConfig, containerName)
if err != nil {
	return "", fmt.Errorf("ERROR: couldn't create container %s\n%+v", containerName, err)
}
```
after this line and it would work in any case.